torch/distributed/distributed_c10d.py

```diff
@@ -3767,3 +3802,50 @@ def new_subgroups_by_enumeration(
                 logger.info("Rank {} is assigned to subgroup {}".format(rank, ranks))
 
     return cur_subgroup, subgroups
+
+
+def _try_find_pg_by_ranks_and_tag(tag: str, ranks: List[int]) -> ProcessGroup:
+    if len(tag) > 0 and not tag.startswith("ptd:") and not tag.startswith("user:"):
+        tag = f"user:{tag}"
+
+    for group in _world.tags_to_pg.get(tag, []):
+        group_ranks = get_process_group_ranks(group)
+        good = all(r in group_ranks for r in ranks)
```

Comment:
what if ranks = [0, 1, 2, 3] but group_ranks = [0, 1, 2, 3, 4, 5, 6, 7]?

good would still be true, but it's not essentially the group we want?

Reply:
good catch!